Pull request

pinchen-fan opened the pull request:
Updated the codes so that they work under Python 3. Fixed a few issues and added some new functions (e.g. calculations of dMAP of the "center of masses: between skymaps.

pinchen-fan committed:
Update antenna-averages

pinchen-fan committed:
Update compare-maps

pinchen-fan committed:
Update plot-maps

pinchen-fan committed:
Update snglFITShtml

pinchen-fan committed:
Delete html-1.16.tar.gz

pinchen-fan committed:
Create ReadMe

pinchen-fan committed:
Add files via upload

pinchen-fan committed:
Update stats.py

pinchen-fan committed:
Update stats.py

pinchen-fan committed:
Update antenna.py

pinchen-fan committed:
Update cartesian.py

pinchen-fan committed:
Update colors.py

pinchen-fan committed:
Update mollweide.py

pinchen-fan committed:
Update mollweide.py

pinchen-fan committed:
Update stats.py

pinchen-fan committed:
Update stats.py

pinchen-fan committed:
Update stats.py

pinchen-fan committed:
Update fits2html.py

reedessick commented:
@pinchen-fan, thanks. This all looks fine to me. I do have a few quick questions, though

  1) have you checked whether this remains compatible with Python 2? Or will the code only be able to run with Python 3 after these changes are made?
  2) the differences in stats.py seem to involve a lot of changes to white space, often on lines that contain no other text. Was there an issue with the indentation in the previous version? Or can you explain what's going on there a bit more?

thanks again

pinchen-fan commented:
@reedessick Thanks. For your questions:
1. Under Python 2.7, I can't install the ligo.skymap package. I only did work in Python 3 environment, and the codes work perfectly there. I just did a quick try with Python 2.7. It seems that line-of-sight projections do not work on the skymaps. Given the limited sample and functions I tried, I think it would be reasonable to suggest that some other functions may be lost, too. Anyway, the only difference is that I couldn't install the ligo.skymap package with Python 2. If you can install it, I feel that the codes should work properly.
2. There were issues with indentations in stats.py. I changed them to tabs only.

One additional thing to note is that heatmaps are not supported, so the sngFITShtml function does not generate skymaps properly.

reedessick commented:
Hi @pinchen-fan,

I'm not particularly comfortable accepting this merge request if it does not update all functionality to be compatible with Python 3. One could quickly find themselves in a situation where the code would not work as designed in either Python 2.7 or Python 3 if only part of the library was updated. As such, I think we should keep your code in a separate branch (or fork) until you can demonstrate that all the functions have been updated appropriately.

cheers

pinchen-fan commented:
Hi @reedessick 

I believe that the issues were already present in your previous codes. The warning you had in your github repo should be the only functions not working.

"WARNING: heatmaps with projection="mollweide" appear to be buggy at this time. This appears to be associated with actually rendering the plot rather than any issue with rotating maps or coordinate transformations. It is currently under investigation but, until it can be resolved, we recommend only examining contours with projection="mollweide" (for some reason, those are not buggy, which suggests the issue is within lalinference.plot.healpix_heatmap)."

I did try the heatmap with Python 2 and it didn't work. It now doesn't work with Python 3, either, so that should be an issue for both versions of Python. 

I do want to emphasize that I updated the code whenever I encountered a problem with multFITShtml or compare-maps. As such, there could be issues in other functions that multFITShtml and compare-maps do not use. 

Also, I changed the cartesian.py file to make time-delay calculations quicker. For lines 425 to 436 in cartesian.py, I think that it might be worth merging that part if you think the calculations are correct. The new calculations that I wrote in there speed up the time-delay process by around 4 times quicker. Time-delay plots and results look the same in the multFITShtml output.

Thanks.
Pinchen

pinchen-fan committed:
Update antenna-averages

pinchen-fan committed:
Update alert2html

pinchen-fan committed:
Update analyze-maps

pinchen-fan committed:
Update compare-maps

pinchen-fan committed:
Update graceids2html

pinchen-fan committed:
Update midVeto

pinchen-fan committed:
Update multFITShtml

pinchen-fan committed:
Update plot-dt

pinchen-fan committed:
Update plot-maps

pinchen-fan committed:
Update snglFITShtml

pinchen-fan committed:
Update fits2html.py

pinchen-fan committed:
Update compare-maps

pinchen-fan committed:
Update compare-maps

pinchen-fan committed:
Update ReadMe

pinchen-fan committed:
Delete Html-1.16.tar.gz

pinchen-fan committed:
Add files via upload

reedessick commented on bin/compare-maps:
```diff
@@ -137,7 +137,7 @@ for ind, label1 in enumerate(labels):
             t2, p2 = d2['estang']
             messages.append( "dtheta_MAP : %.5f %s"%(angle_conversion*np.arccos(stats.cos_dtheta(t1, p1, t2, p2, safe=True)), unit) )
-
+	
```
Please remove this tab/white space

reedessick commented on skymap_statistics/plotting/mollweide.py:
```diff
@@ -23,9 +23,10 @@
 from matplotlib.ticker import Formatter, FixedLocator
 from matplotlib import patheffects
-from matplotlib.projections import projection_registry
+#from matplotlib.projections.__init__ import projection_registry
```
please remove this line since it is commented out anyway

reedessick commented on skymap_statistics/plotting/mollweide.py:
```diff
@@ -265,7 +265,7 @@ def healpix_contour(post, ax, *args, **kwargs):
     y = np.linspace(*ax.get_ylim(), num=500)
     xx, yy = np.meshgrid(x, y)
-
+    
```
please remove this white space

reedessick commented on skymap_statistics/stats.py:
```diff
@@ -16,7 +16,7 @@ def cos_dtheta(theta1, phi1, theta2, phi2, safe=False):
 	"""
 	cosdtheta = np.cos(theta1)*np.cos(theta2) + np.sin(theta1)*np.sin(theta2)*np.cos(phi1-phi2)
-
+	
```
please remove this white space

reedessick commented on skymap_statistics/stats.py:
```diff
@@ -26,7 +26,7 @@ def cos_dtheta(theta1, phi1, theta2, phi2, safe=False):
 			elif cosdtheta > 1:
 				cosdtheta = 1
-
+	
```
please remove this white space

reedessick commented on skymap_statistics/stats.py:
```diff
@@ -138,9 +153,9 @@ def min_cos_dtheta(posterior, theta, phi, nside=None, nest=False, safe=False):
 	## get all ang included in this area
 	thetas, phis = hp.pix2ang(nside, np.arange(len(posterior))[posterior >= posterior[ipix]])
-
+	
```
please remove this white space

reedessick commented on skymap_statistics/stats.py:
```diff
@@ -138,9 +153,9 @@ def min_cos_dtheta(posterior, theta, phi, nside=None, nest=False, safe=False):
+	
 	t, p = estang(posterior)
-
+	
```
please remove this white space

reedessick commented on skymap_statistics/stats.py:
```diff
@@ -165,7 +180,7 @@ def min_all_cos_dtheta_fast(pix, nside, nest=False, safe=False):
 	does this with a boarder-to-boarder search after checking some other things
 	this could be in error up to the pixel size (hopefully small)
-
+	
```
please remove this white space

reedessick commented on skymap_statistics/stats.py:
```diff
@@ -182,56 +197,50 @@ def min_all_cos_dtheta_fast(pix, nside, nest=False, safe=False):
 	if np.sum( selected*antipodes ): ### point and it's antipode are in the set
 		return -1 ### could be in error by the pixel size...
-
+	
```
please remove this white space

reedessick commented on skymap_statistics/stats.py:
```diff
@@ -261,24 +270,24 @@ def __into_modes(nside, pix, nest=False):
 	### establish an array representing the included pixels
 	npix = hp.nside2npix(nside)
-
+	
```
please remove whitespace

reedessick commented on skymap_statistics/stats.py:
```diff
@@ -261,24 +270,24 @@ def __into_modes(nside, pix, nest=False):
 	truth = np.zeros((npix,),bool)
 	truth[pix] = True
-
+	
```
please remove whitespace

reedessick commented on skymap_statistics/stats.py:
```diff
@@ -261,24 +270,24 @@ def __into_modes(nside, pix, nest=False):
+	
 	pixnums = np.arange(npix) ### convenient array we establish once
-
+	
```
please remove whitespace

reedessick commented on skymap_statistics/stats.py:
```diff
@@ -261,24 +270,24 @@ def __into_modes(nside, pix, nest=False):
 		mode = [ipix]
 		to_check = [ipix] ### add it to the list of things to check
-
+		
```
please remove whitespace

reedessick commented on skymap_statistics/stats.py:
```diff
@@ -261,24 +270,24 @@ def __into_modes(nside, pix, nest=False):
 		while len(to_check): # there are pixels in this mode we have to check
 			ipix = to_check.pop() # take one pixel from those to be checked.
-
+			
```
please remove whitespace

reedessick commented on skymap_statistics/stats.py:
```diff
@@ -261,24 +270,24 @@ def __into_modes(nside, pix, nest=False):
+			
 			for neighbour in hp.get_all_neighbours(nside, ipix, nest=nest):# get neighbors as rtheta, rphi
-
+				
```
please remove whitespace

reedessick commented on skymap_statistics/stats.py:
```diff
@@ -290,7 +299,7 @@ def __into_modes(nside, pix, nest=False):
 					pass 
 		modes.append( mode )
-
+	
```
please remove whitespace

reedessick commented on skymap_statistics/stats.py:
```diff
@@ -396,7 +405,7 @@ def symmetric_KLdivergence_walk( posterior1, posterior2, base=2.0, nside=False,
 		posterior2 = resample(posterior2, nside, nest=nest)
 		symKL = symmetric_KLdivergence( posterior1, posterior2, base=base )
-
+		
```
please remove whitespace

reedessick commented on skymap_statistics/stats.py:
```diff
@@ -465,56 +471,48 @@ def __fits_to_table(posterior):
-
 	kde_bandwidth = (pixarea1+pixarea2) ### variance of KDE Gaussian
-
+	
```
please remove whitespace

reedessick commented on skymap_statistics/stats.py:
```diff
@@ -525,86 +523,84 @@ def twoPt_fitstable(posterior, table, Nsamp=101):
 	assumes both are tabular data formats
 	tables must be a numpy array of the form: [(theta, phi, weight), (theta, phi, weight), ...]
-
+	
```
please remove whitespace

reedessick commented on skymap_statistics/stats.py:
```diff
@@ -525,86 +523,84 @@ def twoPt_fitstable(posterior, table, Nsamp=101):
 	t1, p1, posterior1 = table1.transpose()
 	t2, p2, posterior2 = table2.transpose()
-
+	
```
please remove whitespace

reedessick commented on skymap_statistics/stats.py:
```diff
@@ -525,86 +523,84 @@ def twoPt_fitstable(posterior, table, Nsamp=101):
 	kde_bandwidth is the variance used in the Gaussian KDE
-
-        WARNING: this builds a BIG matrix to try to compute this quickly and may be very memory intensive...
+	
```
please remove whitespace

reedessick commented on skymap_statistics/stats.py:
```diff
@@ -525,86 +523,84 @@ def twoPt_fitstable(posterior, table, Nsamp=101):
+	estimate the 2-pt correlation function between 2 posteriors
+	assumes both are tabular data formats
+	tables must be a numpy array of the form: [(theta, phi, weight), (theta, phi, weight), ...]
+	
```
please remove whitespace

reedessick commented on skymap_statistics/stats.py:
```diff
@@ -525,86 +523,84 @@ def twoPt_fitstable(posterior, table, Nsamp=101):
 	IND1 = IND1.flatten()
 	IND2 = IND2.flatten()
-
+	
```
please remove whitespace

reedessick commented on skymap_statistics/stats.py:
```diff
@@ -525,86 +523,84 @@ def twoPt_fitstable(posterior, table, Nsamp=101):
-        dTheta = np.arccos( cos_dtheta(t1[IND1], p1[IND1], t2[IND2], p2[IND2]) )
-
+	dTheta = np.arccos( cos_dtheta(t1[IND1], p1[IND1], t2[IND2], p2[IND2]) )
+	
```
please remove whitespace

reedessick commented on skymap_statistics/stats.py:
```diff
@@ -525,86 +523,84 @@ def twoPt_fitstable(posterior, table, Nsamp=101):
-        count = np.sum(kernal * weights, axis=0) ### sum over all pixel pairs
-
-        return theta, count
+	
```
please remove whitespace

reedessick commented on skymap_statistics/stats.py:
```diff
@@ -525,86 +523,84 @@ def twoPt_fitstable(posterior, table, Nsamp=101):
+	
+	kernal = (2*np.pi*kde_bandwidth)**-0.5 * np.exp(-0.5*(np.outer(dTheta, np.ones_like(theta)) - np.outer(np.ones_like(dTheta), theta))**2/kde_bandwidth)
+	weights = np.outer(posterior1[IND1]*posterior2[IND2], np.ones_like(theta))
+	
```
please remove whitespace

reedessick commented on skymap_statistics/stats.py:
```diff
@@ -525,86 +523,84 @@ def twoPt_fitstable(posterior, table, Nsamp=101):
+	weights = np.outer(posterior1[IND1]*posterior2[IND2], np.ones_like(theta))
+	
+	count = np.sum(kernal * weights, axis=0) ### sum over all pixel pairs
+	
```
please remove whitespace

reedessick commented:
@pinchen-fan, as you can see, I've added a few comments with suggestions for a bit of clean-up. Most of these are associated with tabs/indentations for empty lines. Those are highlighted for you if you look at the "Files changed" tab.

That being said, everything looks fine to me. Thanks for working on this! If you don't want to go through and change the whitespace issues I flagged, just let me know. We can merge this even if you don't want to do that.

Just let me know. We can merge this whenever you'd like.

pinchen-fan committed:
Update compare-maps

pinchen-fan committed:
Update mollweide.py

pinchen-fan committed:
Update compare-maps

pinchen-fan committed:
Update stats.py

pinchen-fan commented:
@reedessick I have made the edits. I just want to make sure that, in empty lines, there should be no white spaces. Correct?

reedessick commented on skymap_statistics/stats.py:
```diff
@@ -62,12 +61,12 @@ def __to_cumulative(posterior):
+	cum = np.empty_like(posterior)
+	cum[ranking] = np.cumsum( posterior[ranking] ) ### and assign it to the correct indecies within cum
+	return cum
+ 	
```
maybe remove this white space too?

reedessick commented on bin/compare-maps:
```diff
@@ -27,6 +25,8 @@ parser.add_option("-d", "--degrees", default=False, action="store_true")
 parser.add_option("-r", "--rankmap", default=False, action="store_true", help="make comparisons with rankmaps instead of probability maps")
 
 parser.add_option("", "--dMAP", default=False, action="store_true", help="compute the angular separation between maximum a posteriori points")
+parser.add_option("", "--dMAP-cm", default=False, action="store_true", help="compute the angular separation between the maximum a posteriori point and the center of mass of the posterior")
```
I don't think this option is ever used. Did you mean to add a conditional below?

reedessick commented:
@pinchen-fan, thanks!

That is correct. My preference is for there to be no spaces or tabs or any type of white space if there is nothing on a line.

I've also added a few more comments. In particular, it looks like you added an option to `compare-maps` that is never actually used. Did you mean to add another conditional based on that?

pinchen-fan committed:
Delete Html-1.16.tar.gz

pinchen-fan committed:
Delete ReadMe

pinchen-fan committed:
Create Html.py

pinchen-fan committed:
Update fits2html.py

pinchen-fan committed:
Update compare-maps
